run.go

```diff
@@ -106,6 +106,37 @@ func doRun(ctx *cli.Context) error {
 
 	log.Infof("loading vms")
 	for _, vm := range suite.Machines {
+
+		if vmConns, ok := suite.Connections[vm.Name]; ok {
+			for nicID, networkName := range vmConns {
+				found := false
+				for netidx := range suite.Networks {
+					network := suite.Networks[netidx]
+					if network.Name == networkName {
+						found = true
+						break
+					}
+				}
+				if !found {
+					return fmt.Errorf("Connection specified unknown network: %s", networkName)
+				}
+				for nidx := range vm.Nics {
+					found = false
```

Comment:
This doesn't look right.  Did you mean the 'found = false' to happen outside of the for nidx := range vm.Nics loop?  Or am I misunderstanding the flow?

It looks like if vm.Nics is empty, you *should* return an error for the connection nic being undefined, but in fact 'found' will be true at line 134 and you'll proceed.

Reply:
I'll review.  I didn't test with a nic-less VMs, so that's likely a bug I'll fix.